src/module/functions/mod.rs

```diff
@@ -418,6 +428,124 @@ impl Module {
 
         Ok(())
     }
+
+    /// Retrieve the ID for the first exported memory.
+    ///
+    /// This method does not work in contexts with [multi-memory enabled](https://github.com/WebAssembly/multi-memory),
+    /// and will error if more than one memory is present.
+    pub fn get_memory_id(&self) -> Result<MemoryId> {
+        if self.memories.len() > 1 {
+            bail!("multiple memories unsupported")
+        }
+
+        self.memories
+            .iter()
+            .next()
+            .map(Memory::id)
+            .context("module does not export a memory")
+    }
+
+    /// Replace a single exported function with the result of the provided builder function.
+    ///
+    /// When called, if `builder` produces a None value, the function in question will be
+    /// replaced with a stub that does nothing (more precisely, a function with an unreachable body).
+    ///
+    /// This function returns the function ID of the *new* function
+    pub fn replace_exported_func_by_id<F>(
+        &mut self,
+        fid: FunctionId,
+        fn_builder: F,
+    ) -> Result<FunctionId>
+    where
+        F: FnOnce((&FuncParams, &FuncResults)) -> Result<Option<LocalFunction>>,
+    {
+        match (self.exports.get_exported_func(fid), self.funcs.get(fid)) {
+            (
+                Some(exported_fn),
+                Function {
+                    kind: FunctionKind::Local(lf),
+                    ..
+                },
+            ) => {
+                // Retrieve the params & result types for the exported (local) function
+                let ty = self.types.get(lf.ty());
+                let (params, results) = (ty.params().to_vec(), ty.results().to_vec());
+
+                // Run the builder function to produce a new local function, or create a stub that is unreachable
+                let new_local_fn = match fn_builder((&params, &results))
+                    .context("export fn builder failed")?
+                {
+                    Some(f) => f,
+                    None => {
+                        let mut builder = FunctionBuilder::new(&mut self.types, &params, &results);
+                        builder.func_body().unreachable();
+                        builder.local_func(vec![])
```

Comment:
Again, I think we should keep stubbing to being an implementation detail.

Reply:
Thanks, will change it here as well

- [ ] remove stubbing